### ISSUE TITLE
Add Custom Edition support into retail

### ISSUE THREAD

**Issue by Kavawuvi** [Kavawuvi/Chimera#1]
This will require:
- Loading Custom Edition maps in retail (fairly easy)
- Joining Custom Edition games in retail (very difficult)

**Comment by Kavawuvi** [comment]
For joining Custom Edition games using stock retail maps, the stun of the following .damage_effect tags will have to be set to 0 to prevent desyncing:
- `vehicles/banshee/banshee bolt.damage_effect`
- `vehicles/ghost/ghost bolt.damage_effect`

To prevent users from joining retail servers with Custom Edition maps, you will not be allowed to use any Custom Edition maps that share the same name as a stock map.

Also, for using existing Custom Edition maps, the ting randomization gain will have to be set to 0.2 due to retail playing this sound 5x as loud as on Custom Edition.

**Comment by Kavawuvi** [comment]
Upon testing, I've found that there are several multiplayer Halo Custom Edition maps hosted on the HAC2 repository that have non-indexed bitmap and sound tags that directly reference the bitmaps.map and sounds.map files.

I always assumed that tool.exe did not do this with multiplayer tags, but it seems that tool.exe actually checks the tag data and raw data separately. Here is the observed behavior:
- If the **tag data is the same size** but the **asset data is a different size**, the **raw data will be internalized** into the map.
- If the **tag data is a different size** but the **asset data is the same size**, the **raw data will be referenced externally** by the tag.
- If the **tag data is the same size** and the **asset data is the same size**, the **tag will be indexed**.

I imagine that this can happen if sprites are added, such as new weapon icons, but some of the bitmaps are the same size. I've observed this on the semi-refined multiplayer maps.

Unfortunately, there is no reliable way to determine which kind of bitmaps.map and sounds.map resources were used when creating these maps, and there may be many maps that have this problem.
